src/vsr/client.zig

```diff
@@ -196,6 +200,48 @@ pub fn Client(comptime StateMachine_: type, comptime MessageBus: type) type {
             if (self.request_timeout.fired()) self.on_request_timeout();
         }
 
+        /// Registers a session with the cluster for the client, if this has not yet been done.
+        pub fn register(self: *Self, callback: Request.RegisterCallback, user_data: u128) void {
+            assert(self.request_inflight == null);
+            assert(self.request_number == 0);
+
+            const message = self.get_message().build(.request);
+            errdefer self.release_message(message.base());
+
+            // We will set parent, session, view and checksums only when sending for the first time:
+            message.header.* = .{
+                .size = @sizeOf(Header) + @sizeOf(vsr.RegisterRequest),
```

Comment:
Do we have some smoke test that sending just `@sizeOf(Header)` bytes works?

I guess its a medium-size architecture question of how we could add such a test, as, presumably, we don't want to support it explicitly in client's code, but we still want to use client in the simulator...

Reply:
Yeah, I think this comes down to how we plan to test multi-version compatibility (for both server+client). :thinking: